jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -927,16 +931,166 @@ static String formatExpires(Instant expires)
     }
 
     /**
-     * <p>Parses the {@code Expires} attribute value
-     * (in RFC 1123 format) into an {@link Instant}.</p>
+     * <p>Parses the {@code Expires} attribute value using algorithm
+     * specified in <a href="https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.1">RFC6265: Section 5.1.1: Date</a>
+     * in into an {@link Instant}.</p>
      *
-     * @param expires an instant in the RFC 1123 string format
+     * @param expires an instant in the string format supported by RFC6265 HTTP State Management
      * @return an {@link Instant} parsed from the given string
      */
     static Instant parseExpires(String expires)
     {
-        // TODO: RFC 1123 format only for now, see https://www.rfc-editor.org/rfc/rfc2616#section-3.3.1.
-        return ZonedDateTime.parse(expires, DateTimeFormatter.RFC_1123_DATE_TIME).toInstant();
+        Objects.requireNonNull(expires, "Expires string cannot be null");
+
+        int year = -1;
+        int month = -1;
+        int day = -1;
+        int hour = -1;
+        int minute = -1;
+        int second = -1;
+
+        // delimiter       = %x09 / %x20-2F / %x3B-40 / %x5B-60 / %x7B-7E
+        // Note: That leaves "0-9" + "A-Z" + "a-z" + ":"
+        QuotedStringTokenizer dateTokenizer = QuotedStringTokenizer.builder()
```

Comment:
Should this be a Quoted tokenize? Perhaps we simply want to split here?
Eitherway, would be good not to have to build a new tokenizer on every call and instead either have one shareable static or a `ThreadIdPool` of them.

Reply:
Turns out String.split() is very slow in comparison to the QST, went from 1,400,00 ops/s to 540,000 ops/s simply by using String.split()

Reply:
Try Pattern.split on a statically compiled regex pattern

Reply:
From commit ae52f81d6ec5ab093d109e67b15c44de42f3f5a6 using QuotedStringTokenizer

```
Benchmark                                            Mode  Cnt        Score       Error  Units
HttpCookieParseExpireBenchmark.testParse            thrpt   10  1613127.922 ± 47840.084  ops/s
HttpCookieParseExpireBenchmark.testParseExpiresOld  thrpt   10   678048.189 ± 25679.250  ops/s
```

From commit c494a8f300e775ca5dddb3a69c01236ee2052978 using Pattern.split() on a precompiled Pattern.

```
Benchmark                                            Mode  Cnt       Score       Error  Units
HttpCookieParseExpireBenchmark.testParse            thrpt   10  678636.962 ± 31548.239  ops/s
HttpCookieParseExpireBenchmark.testParseExpiresOld  thrpt   10  689252.488 ± 23301.442  ops/s
```

I'm going to revert that last commit.

Reply:
Surprisingly, if I switch to a `new java.util.StringTokenizer` for every parse, the performance is even better.

This is from commit f59dcb1075988e66364879c4fd3bc6a3f484f3d9

```
Benchmark                                            Mode  Cnt        Score       Error  Units
HttpCookieParseExpireBenchmark.testParse            thrpt   10  2213757.822 ± 71140.452  ops/s
HttpCookieParseExpireBenchmark.testParseExpiresOld  thrpt   10   693448.992 ± 24155.550  ops/s
```

Reply:
Then that will do!   The class is deprecated but easy enough to replace if it ever goes away.